cloudfoundry/provider/types_app.go

```diff
@@ -453,8 +496,21 @@ func mapAppValuesToType(ctx context.Context, appManifest *cfv3operation.AppManif
 		appType.Routes = types.SetNull(routeObjType)
 	}
 	if appManifest.Env != nil {
-		appType.Environment, tempDiags = types.MapValueFrom(ctx, types.StringType, appManifest.Env)
-		diags = append(diags, tempDiags...)
+		// If we have the original plan, use it as the base to preserve sensitivity markings
```

Comment:
I do not understand the extensive refactoring of the code here that is not related to the intent of the PR focusing on lifecycle (or am I missing something).